pandas/core/ops/array_ops.py

```diff
@@ -132,3 +152,177 @@ def na_arithmetic_op(left, right, op, str_rep, eval_kwargs):
         result = masked_arith_op(left, right, op)
 
     return missing.dispatch_fill_zeros(op, left, right, result)
+
+
+def arithmetic_op(left, right, op, str_rep, eval_kwargs):
+
+    from pandas.core.ops import (
+        maybe_upcast_for_op,
+        should_extension_dispatch,
+        dispatch_to_extension_op,
+    )
+
+    keep_null_freq = isinstance(
+        right,
+        (
+            ABCDatetimeIndex,
+            ABCDatetimeArray,
+            ABCTimedeltaIndex,
+            ABCTimedeltaArray,
+            Timestamp,
+        ),
+    )
+
+    # NB: We assume that extract_array has already been called on `left`, but
+    #  cannot make the same assumption about `right`.  This is because we need
+    #  to define `keep_null_freq` before calling extract_array on it.
+    lvalues = left
+    rvalues = extract_array(right, extract_numpy=True)
+
+    rvalues = maybe_upcast_for_op(rvalues, lvalues.shape)
+
+    if should_extension_dispatch(left, rvalues) or isinstance(
+        rvalues, (ABCTimedeltaArray, ABCDatetimeArray, Timestamp)
+    ):
+        # TimedeltaArray, DatetimeArray, and Timestamp are included here
+        #  because they have `freq` attribute which is handled correctly
+        #  by dispatch_to_extension_op.
+        res_values = dispatch_to_extension_op(op, lvalues, rvalues, keep_null_freq)
+
+    else:
+        with np.errstate(all="ignore"):
+            res_values = na_arithmetic_op(lvalues, rvalues, op, str_rep, eval_kwargs)
+
+    return res_values
+
+
+def comparison_op(left, right, op):
+    from pandas.core.ops import should_extension_dispatch, dispatch_to_extension_op
+
+    # NB: We assume extract_array has already been called on left and right
+    lvalues = left
+    rvalues = right
+
+    rvalues = lib.item_from_zerodim(rvalues)
```

Comment:
This isn't done yet higher up? (arithmetic_op is not doing this?)

Reply:
arithmetic_op doesnt, but probably should.  Things like this are the motivation for #28037.